Topic 0: Introduction to OOP Concepts

hherminioandres opened the pull request:
- Created the singleton pattern example for the DB connection.
- Created a abstract factory pattern example for different type of SQL connections.
- Created the proxy pattern for DB accesor clases.
- Created a builder pattern example for DB connection.

hherminioandres committed:
Topic 0: Introduction to OOP Concepts

- Created the singleton pattern example for the DB connection.
- Created a abstract factory pattern example for different type of SQL connections.
- Created the proxy pattern for DB accesor clases.
- Created a builder pattern example for DB connection.

salaesp commented on .classpath:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" path="src"/>
```
please add IDE files to the .gitignore

salaesp commented on src/com/AbstractFactory/SQLConnectionFactory.java:
```diff
@@ -0,0 +1,19 @@
+		return null;
+	}
+	
+	public static void main(String[] args) {
```
you should have only one main per app, this is the application entry point.

hherminioandres committed:
Topic 0 correction

-The second main was deleted
-IDE files were added to .gitignore

hherminioandres committed:
The IDE files were deleted

carlos-lopez-glb commented on src/com/AbstractFactory/AbstractFactory.java:
```diff
@@ -0,0 +1,27 @@
+public class AbstractFactory {
+	
+	public SQLFactory getConnection(String acc) {
+		if(acc.equalsIgnoreCase("JDBC"))
```
Put those hardcoded values as Constants

carlos-lopez-glb replied:
a good practice is to include brackets even if not completely necessary to improve readability

carlos-lopez-glb commented on src/com/AbstractFactory/AbstractFactory.java:
```diff
@@ -0,0 +1,27 @@
+	public SQLFactory getConnection(String acc) {
+		if(acc.equalsIgnoreCase("JDBC"))
+			return new JDBC();
+		if(acc.equalsIgnoreCase("OLEDB"))
```
The equals comparison is better to do in the inverse way
`if (OLEDB.equalsIgnoreCase(acc))`
This is **Null Safety** because the parameter (acc) can be null

hherminioandres committed:
Topic 0 correction

- The Constant class was added to not use hardcoded.
- A good practice included: brackets
- Null safety - Changed the equals comparison way.

carlos-lopez-glb commented on src/com/AbstractFactory/OLEDB.java:
```diff
@@ -0,0 +1,14 @@
+
+public class OLEDB extends SQLFactory {
+	
+	static final String SQLSERVER = "SQLSEVER";
```
This constant is not used, you are using Constant.SQLSERVER. Please remove it if not used.
Anyway is not common create a separate class for constants, constants are usually defined in the class it's used.

hherminioandres committed:
Correction of constant variables

The Constant class was removed from this project and each class has the constants that it uses
